### PR TITLE
Tooltips

### PR DESCRIPTION
**Pull request by Tyroche**
I added some tooltips to the info boxes, still isn't perfected for the right hand view (in character creation).

Should work for any shop though!  The styling is all subjective, so feel free to go change it all if it doesn't look good :).

### CODE REVIEW

**Comment by etkirsch** [comment]
This looks good but before merge I'd like to get your thoughts on replacing the switch statement.

```javascript
switch(item.rarity){
  case "Common":
    rarity = '<hr/><span class="rarity-commmon">Common</span><hr/>';
    break;
  case "Uncommon":
    rarity = '<hr/><span class="rarity-uncommmon">Uncommon</span><hr/>';
    break;
  case "Rare":
    rarity = '<hr/><span class="rarity-rare">Rare</span><hr/>';
    break;
  case "Very Rare":
    rarity = '<hr/><span class="rarity-very-rare">Very Rare</span><hr/>';
    break;
  case "Legendary":
    rarity = '<hr/><span class="rarity-legendary">Legendary</span><hr/>';
    break;
  case "Artifact":
    rarity = '<hr/><span class="rarity-artifact">Artifact</span><hr/>';
    break;
  case "Unique":
    rarity = '<hr/><span class="rarity-unique">Unique</span><hr/>';
    break;
  default:
    rarity = '<hr/><span class="rarity-commmon">Unknown</span><hr/>';
    break;
}
```

to... 

```javascript
// Rarity will always default to Common thanks to the Schema
var style = "rarity-" + item.rarity.toLowerCase();

return '<hr><span class="' + style + '">' + item.rarity + '</span><hr>';
```

Unfortunately this isn't using ES6 so we don't get the fancy string interpolation but that'll do. Another thing I noticed is that you may need to check your styles, looks like 'rarity-commmon' has three ms lol

**Comment by etkirsch** [comment]
(I'm innately prejudiced against 'else' and 'switch' statements, if you haven't noticed)

**Comment by Tyroche** [comment]
No problem!

I don't like switch statements either, so this is a welcome suggestion :)

**Comment by Tyroche** [comment]
Made the changes!  

In case you were wondering, I had to split out the rendering and the display logic because tooltip-html only taking context-safe strings, and I couldn't get angular to recompile the sanitized string :(.

I think this is fine though -- gets the job done!

File: modules/items/client/directives/shop-list.client.directive.js [hunk 1]
```diff
@@ -59,6 +60,19 @@
         }
       }
 
+      function renderTooltip(item){
```
Comment (NullVoxPopuli) [inline modules/items/client/directives/shop-list.client.directive.js]: why is this a method, as apposed to part of a template?

**Comment by etkirsch** [comment]
@NullVoxPopuli with respect to your comment, I think it might be better to use a standard template and inject it in the ng-repeat using `tooltip-template` like shown on [bootstrap's tooltip section](http://angular-ui.github.io/bootstrap/#/tooltip). If we do it in the ng-repeat we won't have to worry about dynamically rendering the tooltip on every mouseover. We also wouldn't need to worry about sanitizing with that.

**Comment by NullVoxPopuli** [comment]
I think I get what you're saying, I meant just have a template that does something ilke the following:
```hbs
<span class='bold-item-name'> {{ itemName }} </span>
<hr>
<span class={{ rarityClass }}> {{ rarity }} </span>
<hr>
{{ description }}
```
given that `rarity`, `rerityClass`, `itemName`, and `description` are computed properties.

This is in the handlebars format.

the tooltip should be hidden by default, and css should be able to show it upon hover, which I think would address any rendering performance issues.

**Pull request merged**